<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1143</width>
    <height>585</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="autoFillBackground">
   <bool>false</bool>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>270</x>
      <y>110</y>
      <width>47</width>
      <height>13</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QWidget" name="Input_widget" native="true">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>611</width>
      <height>341</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <widget class="QPushButton" name="choose_data_btn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>0</y>
       <width>81</width>
       <height>41</height>
      </rect>
     </property>
     <property name="text">
      <string>Choose data</string>
     </property>
    </widget>
    <widget class="QTableView" name="tableView">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>271</width>
       <height>291</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="data_path">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>10</y>
       <width>381</width>
       <height>21</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>50</y>
       <width>61</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Input value</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>90</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Value of k</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>110</y>
       <width>91</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Number of clusters</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="k_text">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>80</y>
       <width>113</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="c_text">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>110</y>
       <width>113</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="epsilon_text">
     <property name="geometry">
      <rect>
       <x>400</x>
       <y>140</y>
       <width>113</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>140</y>
       <width>47</width>
       <height>13</height>
      </rect>
     </property>
     <property name="text">
      <string>Epsilon</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="m_text">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>190</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
     <property name="text">
      <string/>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>290</x>
       <y>190</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>m</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="M_text">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>190</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>190</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>M</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="M1_text">
     <property name="geometry">
      <rect>
       <x>450</x>
       <y>220</y>
       <width>71</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>420</x>
       <y>220</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>M'</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="algo1_check">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>170</y>
       <width>81</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Algorithm 1</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="algo2_check">
     <property name="geometry">
      <rect>
       <x>430</x>
       <y>170</y>
       <width>81</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>Algorithm 2</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_2pha_algo2">
     <property name="geometry">
      <rect>
       <x>430</x>
       <y>250</y>
       <width>70</width>
       <height>17</height>
      </rect>
     </property>
     <property name="text">
      <string>2 Pha ?</string>
     </property>
    </widget>
    <widget class="QCheckBox" name="check_2pha_algo1">
     <property name="geometry">
      <rect>
       <x>310</x>
       <y>250</y>
       <width>70</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>2 Pha ?</string>
     </property>
    </widget>
    <widget class="QPushButton" name="run_btn">
     <property name="geometry">
      <rect>
       <x>360</x>
       <y>290</y>
       <width>75</width>
       <height>23</height>
      </rect>
     </property>
     <property name="text">
      <string>RUN</string>
     </property>
    </widget>
   </widget>
   <widget class="QTabWidget" name="M_table">
    <property name="geometry">
     <rect>
      <x>710</x>
      <y>10</y>
      <width>341</width>
      <height>341</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <property name="tabsClosable">
     <bool>false</bool>
    </property>
    <property name="movable">
     <bool>true</bool>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>U_ngang</string>
     </attribute>
     <widget class="QTableView" name="U_ngang_table">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>341</width>
        <height>311</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>M</string>
     </attribute>
     <widget class="QTableView" name="tableView_3">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>341</width>
        <height>311</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_3">
     <attribute name="title">
      <string>U</string>
     </attribute>
     <widget class="QTableView" name="U_table">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>341</width>
        <height>311</height>
       </rect>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QTabWidget" name="output_widget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>350</y>
      <width>571</width>
      <height>201</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_4">
     <attribute name="title">
      <string>V_matrix</string>
     </attribute>
     <widget class="QTableView" name="V_table">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>31</y>
        <width>261</width>
        <height>141</height>
       </rect>
      </property>
     </widget>
     <widget class="QTableView" name="V_table_2">
      <property name="geometry">
       <rect>
        <x>300</x>
        <y>30</y>
        <width>261</width>
        <height>141</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>10</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">rgg</string>
      </property>
      <property name="text">
       <string>Algorithm 1</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>10</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Algorithm 2</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_5">
     <attribute name="title">
      <string>Class_Cluster</string>
     </attribute>
     <widget class="QTableWidget" name="V_1_table">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>30</y>
        <width>261</width>
        <height>151</height>
       </rect>
      </property>
     </widget>
     <widget class="QTableWidget" name="V_2_table">
      <property name="geometry">
       <rect>
        <x>300</x>
        <y>30</y>
        <width>261</width>
        <height>151</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>10</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Algorithm 1</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>10</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Algorithm 2</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>600</x>
      <y>370</y>
      <width>551</width>
      <height>191</height>
     </rect>
    </property>
    <property name="autoFillBackground">
     <bool>true</bool>
    </property>
    <widget class="QTableWidget" name="validity_table">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>40</y>
       <width>531</width>
       <height>121</height>
      </rect>
     </property>
     <property name="showGrid">
      <bool>true</bool>
     </property>
     <property name="gridStyle">
      <enum>Qt::SolidLine</enum>
     </property>
     <property name="sortingEnabled">
      <bool>true</bool>
     </property>
     <row>
      <property name="text">
       <string>Rand Index</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>Adjusted Rand Index</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>Jaccard Coefficient</string>
      </property>
     </row>
     <column>
      <property name="text">
       <string>Algorithm1(1pha)</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Algorithm1(2pha)</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Algorithm2(1pha)</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Algorithm2(2pha)</string>
      </property>
     </column>
     <item row="0" column="0">
      <property name="text">
       <string/>
      </property>
     </item>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>10</y>
       <width>101</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>External_validity</string>
     </property>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1143</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>choose_data_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>choose_data()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>41</y>
    </hint>
    <hint type="destinationlabel">
     <x>462</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>run_btn</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>caculate_cluster()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>407</x>
     <y>322</y>
    </hint>
    <hint type="destinationlabel">
     <x>571</x>
     <y>292</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>choose_data()</slot>
  <slot>caculate_cluster()</slot>
 </slots>
</ui>
